sdk/python/kfp/components/pipeline_task.py

```diff
@@ -474,6 +475,28 @@ def my_pipeline():
             self._task_spec.dependent_tasks.append(task.name)
         return self
 
+    def ignore_upstream_failure(self) -> 'PipelineTask':
+        """If called, the pipeline task will run when any specified upstream
+        tasks complete, even if unsuccessful.
+
+        This method effectively turns the caller task into an exit task
+        if the caller task has upstream dependencies.
+        """
+
+        from kfp.components import pipeline_context
+        for input_spec_name, input_spec in self.component_spec.inputs.items():
+            for input_name, argument_value in self._inputs.items():
+                if (input_spec_name == input_name) and (isinstance(
```

Comment:
Can we index into the inputs with `self._inputs[input_spec_name]` to avoid the nested for loop?